### PR TITLE
Send MAVLink DO_DIGICAM_CONTROL on camera trigger

### PR DESCRIPTION
**Pull request by jonrwat**

### CODE REVIEW

**Comment by dagar** [comment]
Isn't the camera already triggered by a command that could instead be forwarded?

**Comment by jonrwat** [comment]
For a MAVLink compliant camera sending of a MAV_CMD_IMAGE_START_CAPTURE is already [implemented](https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_messages.cpp#L1423).  I'm using a camera which triggers based on the MAV_CMD_DO_DIGICAM_CONTROL message though.  I figured a software update to send both packets on a camera trigger was appropriate.

**Comment by dagar** [comment]
This should be fine for now, but ideally we'd want the camera_trigger driver to be configurable for mavlink trigger types, then use that configuration for mavlink publishing.

If you send a DO_DIGICAM_CONTROL from the ground and have you have the camera_trigger set to mavlink output and mavlink command forwarding enabled your camera will receive an extra DO_DIGICAM_CONTROL.

**Comment by DonLakeFlyer** [comment]
I don't understand how those vehicle global camera trigger settings like distance make sense? That seems like a wrong way to do it.

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -1436,6 +1436,23 @@ class MavlinkStreamCameraTrigger : public MavlinkStream
 				msg_cmd.param7 = NAN;
 
 				mavlink_msg_command_long_send_struct(_mavlink->get_channel(), &msg_cmd);
+
+				/* send MAV_CMD_DO_DIGICAM_CONTROL*/
+				mavlink_command_long_t digicam_ctrl_cmd;
+
+				digicam_ctrl_cmd.target_system = mavlink_system.sysid;
```
Comment (zehortigoza) [inline src/modules/mavlink/mavlink_messages.cpp]: So the target system id is the same sending it? This is wrong... use broadcast id instead.

Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp]: Thanks @zehortigoza I wasn't aware you could use 0 for broadcast. I'll make the change.

**Comment by dagar** [comment]
@DonLakeFlyer yes, it likely needs some work. Want to open an issue to discuss or sync up on skype?

File: src/modules/mavlink/mavlink_messages.cpp (hunk 1 above)
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp]: PR for broadcast - https://github.com/PX4/Firmware/pull/6896

**Pull request merged**